### PR TITLE
Make README more inviting

### PR DESCRIPTION
**Pull request by jez**
### Summary & motivation

- Add new **Table of Contents** section
- Add new **Demo** section
- More compliant with [standard-readme](https://github.com/RichardLitt/standard-readme)
- Now with infinitely more screen shots

### Testing & documentation

No code changes => no testing!

[**OLD README**](https://github.com/stripe/react-stripe-elements/blob/de100fd2f1f979e475bfd470e977cae9973cceb2/README.md)
[**NEW README**](https://github.com/stripe/react-stripe-elements/blob/ef761b6c66f526585177f1ab49b4c92f25457645/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,44 +1,97 @@
...
 This project is a thin React wrapper around [Stripe.js](https://stripe.com/docs/stripe.js)
 and [Stripe Elements](https://stripe.com/docs/elements). It allows you to add Elements
 to any React app, and manages the state and lifecycle of Elements for you.
 
 The [Stripe.js / Stripe Elements API reference](https://stripe.com/docs/elements/reference)
 goes into more detail on the various customization options for Elements (e.g. styles, fonts).
 
-## Getting started
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
```
Comment (michelle) [inline README.md]: Could you add instructions for updating the README? (`yarn doctoc`, for example)

Comment (jez) [reply README.md]: 👍

File: README.md [hunk 2]
```diff
@@ -389,6 +442,10 @@ We use [prettier](https://github.com/prettier/prettier) for code formatting:
 
     yarn run prettier
 
+To update the ToC if you make a change to any headers:
```
Comment (michelle) [inline README.md]: "a change to the README" maybe?

headers is vague in this context

**Comment by michelle** [comment]
> Forgot to say "README" in the instructions for updating the README in the README

😂

**Comment by michelle** [comment]
lgtm

**Pull request merged**